source/Calamari/Deployment/PackageRetention/Model/JournalEntry.cs

```diff
@@ -1,19 +1,49 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 using Calamari.Common.Plumbing.Deployment.PackageRetention;
+using Newtonsoft.Json;
 
 namespace Calamari.Deployment.PackageRetention.Model
 {
     public class JournalEntry
     {
         public PackageIdentity Package { get; }
-        public PackageLocks PackageLocks { get; }
-        public PackageUsage PackageUsage { get; }
+        
+        [JsonProperty]
+        readonly PackageUsages usages;
 
-        public JournalEntry(PackageIdentity package, PackageLocks packageLocks = null, PackageUsage packageUsage = null)
+        [JsonProperty]
+        readonly PackageLocks locks;
+
+        [JsonConstructor]
+        public JournalEntry(PackageIdentity package, PackageLocks packageLocks = null, PackageUsages packageUsages = null)
         {
             Package = package ?? throw new ArgumentNullException(nameof(package));
-            PackageLocks = packageLocks ?? new PackageLocks();
-            PackageUsage = packageUsage ?? new PackageUsage();
+            locks = packageLocks ?? new PackageLocks();
+            usages = packageUsages ?? new PackageUsages();
+        }
+
+        public void AddUsage(ServerTaskId deploymentTaskId, CacheAge cacheAge)
+        {
+            usages.Add(new UsageDetails(deploymentTaskId, cacheAge));
+        }
+
+        public void AddLock(ServerTaskId deploymentTaskId, CacheAge cacheAge)
+        {
+            locks.Add(new UsageDetails(deploymentTaskId, cacheAge));
+        }
+
+        public void RemoveLock(ServerTaskId deploymentTaskId)
+        {
+            //We only remove the first lock that matches the deployment task id, in case we have a deployment that uses the same package twice.
```

Comment:
I don't think this lines up with the server PR - the release lock command is sent once per package regardless of how many times it's used in the deployment.

Reply:
Good point, I'll change it.